src/App.js

```diff
@@ -1,16 +1,36 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
+import ChatLog from './components/ChatLog';
 import chatMessages from './data/messages.json';
 
 const App = () => {
+  const [entries, setEntries] = useState(chatMessages);
+
+  const calcTotalLikes = () => {
+    return entries.reduce((total, entry) => {
+      return entry.liked ? total + 1 : total;
+    }, 0);
+  };
+
+  const updateLike = (id) => {
```

Comment:
So nothing is wrong with this code here, I just want to show you best practice: 

```javascript
const updateLike = (id) => {
 setEntries(prevEntries => {
      const newEntries = prevEntries.map(entry => {
        return entry.id === id ? { ...entry, liked: !entry.liked } : entry
      })
      return newEntries
    })
```

The best practice part being `setEntries( prevEntries => {....})` passing a callback to the setter function that `useState` provides allows you to have a copy of your `state` aka the `prevEntries`. Though you are not changing anything with the `map` function this copy you want to work and not directly working with your `entries` data.